### ISSUE TITLE
Unformatted Meta Data Displays in Exceprts

### ISSUE THREAD

**Issue by robneu** [hlashbrooke/Seriously-Simple-Podcasting#34]
If you set the option to display the podcast meta data in the full content and set it display to before the content, unformatted meta data will be passed through to `the_excerpt`. The reason for this is that `wp_trim_excerpt` applies `the_content` filter before setting up the excerpt [here](https://github.com/WordPress/WordPress/blob/d0c327443c9ecf90abf057d1f57648c1a92b042b/wp-includes/formatting.php#L2552).

Unless you get around to fixing it before I wake up, I'll work up a patch for this and send it along tomorrow. Cheers!

**Comment by hlashbrooke** [comment]
Thanks - I was aware of this, but haven't yet worked out an easy way around it (haven't actually looked too hard for a solution). I don't think I'm going to have time to work on this today, so if you haven't submitted something by tomorrow then I'll get it sorted out then. Would be good to have this out in v1.8.3 later this week :)

**Comment by hlashbrooke** [comment]
Having a bit of trouble with this one and I can't quite see a way around it. I've split the function called on `the_content` filter into two separate functions (one for the full content and one for the excerpt) and then they both call an independent function that just gets the audio player and episode meta for the given post ID. This enables me to do an `is_single()` check inside the function hooked onto `the_content`, which works just fine.

The problem, however, is that sometimes the full content is shown on the archive pages (which is actually the default for many of the default WP themes) so the player & meta should be displayed there too, but it obviously fails the `is_single()` check.

I can't really see a way around this seeing as `the_content` is called for both the excerpt and the content. What core really needs here is two separate functions, or at least a context flag to you know where the filter is being used.

Not sure if you know of a better way to handle this that I just haven't thought of - anything would help :)

**Comment by robneu** [comment]
I worked on this for a while yesterday and it's a tricky one! I thought splitting it like that might be a good idea, but there could be some edge cases where people are displaying excerpts somewhere other than archives.

I tried a lot of filter juggling to see if it was possible to handle using `has_filter` but nothing worked as expected. Strangely, checking for filters related to `the_excerpt` removed the output from `the_content` instead of `the_excerpt` O_o 

I even tried changing the meta function into a shortcode and printing that like so:

```php
public function display_content_meta_data( $content ) {
	switch ( get_option( 'ss_podcasting_player_content_location', 'above' ) ) {
		case
			'above': $content = '[ss_podcast_meta]' . $content;
		break;
		case
			'below': $content = $content . '[ss_podcast_meta]';
		break;
	}
	return $content;
}
```
... but that creates more or less the same result. I think splitting the function like you mentioned is probably the only sane method. The only other way would probably be somef crazy preg_replace regex to get rid of the unformatted version. There might be another option, but I'm not seeing it. :\
